src/output-json-alert.c

```diff
@@ -844,6 +844,8 @@ static TmEcode JsonAlertLogThreadDeinit(ThreadVars *t, void *data)
     MemBufferFree(aft->json_buffer);
     MemBufferFree(aft->payload_buffer);
 
+    LogFileDeInitCtx(aft->json_output_ctx->file_ctx, t->id);
```

Comment:
I'm trying to figure out if we can get this out of the loggers.  Unless a plugin is in use, this is a no-op, yet when its a regular file, it manages to get cleaned up without any extra calls here.  So I'm wondering if we can move the same sort of cleanup for plugins out of individual loggers.  I'm tracing through this myself, but any thoughts?

Reply:
Ok, so when using regular files and threaded, the per thread file contexts are never cleaned up. I've created an issue for this: https://redmine.openinfosecfoundation.org/issues/4425

I think this should probably be resolved before dealing with shutdown in plugins.

Reply:
If this is rebased onto https://github.com/OISF/suricata/pull/6051, can this be cleaned up some?